java-checks/src/main/java/org/sonar/java/checks/CombineCatchCheck.java

```diff
@@ -55,6 +58,33 @@ public void visitNode(Tree tree) {
     }
   }
 
+  private static boolean areNotSameMethodCalls(JavaTree leftNode, JavaTree rightNode) {
```

Comment:
Could this be a part of general equivalence check? To my mind, we can't say that 2 blocks are equivalent, if they're look the same but referencing different methods (in general, not only for this rule)

This change might also influence other rules, so need to check.